src/parser/serializer.ts

```diff
@@ -319,6 +320,29 @@ function serializeObjectId(
   return index + 12;
 }
 
+function serializeUUID(buffer: Buffer, key: string, value: UUID, index: number, isArray?: boolean) {
+  // Write the type
+  buffer[index++] = constants.BSON_DATA_UUID;
+  // Number of written bytes
+  const numberOfWrittenBytes = !isArray
```

Comment:
To me it will a bit more readable if the condition is `isArray` (and swap the two branches).

Reply:
The whole `serializeUUID` will be removed in the next iteration (but this is basically copy/pasted from `serializeObjectId`, so it's inline with the existing code).